### PR TITLE
Deprecate old list plugins

### LINKED ISSUE #14942: Depreciate Old `List` Plugins in Favor of `DocumentList` Equivalents
## Provide a description of the task

Part of #14632.

### Scope
1. Rename existing `List*` plugins to `LegacyList*` plugins.
  This change will allow integrators to fall back to the old solution if they encounter any issues after an update.
2. Promote `DocumentList*` Plugins (by renaming them to `List*`)
  This should enable document lists for all integrations. Assuming we achieved feature parity, users should not miss any feature.
3. Support current configurations:
    - Introduce `DocumentList*` plugins that pull the new `List*` plugins as a requirement.
    - If an integration currently uses the `DocumentList`, it should continue to work without necessitating any configuration changes.
    - `console.warn`ing should suggest updating configurations.
4. Completely remove the `ListStyle` plugin. It has been deprecated for some time now.
5. Write migration guide. 
    * Mention the possibility of using `LegacyList` in case of issues.
    * Mention the `multiBlock` config.
 

## 📃 Other details

* Browser: …
* OS: …
* CKEditor version: …
* Installed CKEditor plugins: …

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Other (list): The Document List feature was promoted to the main list feature implementation. The "Document" prefix was removed while the old list implementation was prefixed with the "Legacy" keyword. A compatibility alias plugins were introduced (`DocumentList`, `DocumentListProperties`, `TodoDocumentList`) for backward compatibility but those are marked as deprecated and log a warning in the browser console. Closes #14942.

MINOR BREAKING CHANGE (list): All old list plugins are now prefixed with `legacy` (including directory names):  `List` -> `LegacyList`, `ListProperties` -> `LegacyListProperties`, `TodoList` -> `LegacyTodoList`, `ListEditing` -> `LegacyListEditing`, `ListUtils` -> `LegacyListUtils`, `ListPropertiesEditing` -> `LegacyListPropertiesEditing`, `TodoListEditing` -> `LegacyTodoListEditing`. See #14942.

MINOR BREAKING CHANGE (list): The Document List plugins are now not prefixed with `document` (including directory names): `DocumentList` -> `List`, `DocumentListProperties` -> `ListProperties`, `TodoDocumentList` -> `TodoList`, `DocumentListEditing` -> `ListEditing`, `DocumentListUtils` -> `ListUtils`, `DocumentListPropertiesEditing` -> `ListPropertiesEditing`, `DocumentListPropertiesUtils` -> `ListPropertiesUtils`, `TodoDocumentListEditing` -> `TodoListEditing`. See #14942.

MAJOR BREAKING CHANGE (list): The `AdjacentListsSupport` plugin is moved from the `documentlist` directory to the `list` directory. See #14942.

MINOR BREAKING CHANGE (list): The `ListStyle` plugin was removed since it was deprecated for a while. Use the `ListProperties` plugin instead. See #14942.

---

### Additional information

PR in the commercial repo: https://github.com/cksource/ckeditor5-commercial/pull/5853

### CODE REVIEW

**Comment by niegowski** [comment]
The `removePlugins` option could lead to troubles in integrations. I noticed it in a manual test where it throws:
```
  ❌ CKEditorError: plugincollection-required {"plugin":"List","requiredBy":"DocumentList"}
    Read more: https://ckeditor.com/docs/ckeditor5/latest/support/error-codes.html#error-plugincollection-required
```
it's because it was injecting `DocumentList` and `DocumentListProperties` plugins and passing `removePlugins: [ List, ListProperties, TodoList ]`

File: docs/updating/update-to-41.md [hunk 1]
```diff
@@ -22,12 +22,93 @@ Listed below are the most important changes that require your attention when upg
 
 ### Breaking changes to the list plugin
 
-As of the latest release, the current list plugin (often referred to as list v1.0) has been replaced with the {@link features/lists newer and more advanced list plugin}, formerly known as Document lists (v2.0).
+As of the latest release, the current list plugin (often referred to as list v1.0) has been replaced with the {@link features/document-lists newer and more advanced list plugin}, formerly known as Document lists (v2.0).
```
Comment (godai78) [inline docs/updating/update-to-41.md:25]: Document lists guide doesn't exist anymore.

**Comment by godai78** [comment]
Docs links errors will be fixed with this PR: #14962  
  
They are ommittable as of now.

File: docs/updating/update-to-41.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+The list v2 (document list) feature was implemented in 2022 to add support for block content in list items. It supported extending list markup via GHS. It did not, however, support to-do lists back then. We concentrated on bringing full list v1 functionality to this plugin. The newest release brings in the to-do list functionality and the {@link features/lists-editing#simple-lists simple list} configuration setting.
+
+We introduced the new plugin in a manner that aims to be transparent for our users, namely by physically replacing the old plugin with the new one, but retaining all namespace intact. It means, starting with release v.41.0.0 all imports of various lists-related plugins will use the new version.
+
+The old plugin has been renamed to `LegacyList` instead. The same applies to all other list-related plugins, namely: `LegacyListProperties`, and `LegacyTodoList`.
+
+Unless you need to specifically use the old v1.0 plugin in your integration, there is no need to make changes in the configuration.
```
Comment (niegowski) [inline docs/updating/update-to-41.md:33]: We should mention this here: https://github.com/ckeditor/ckeditor5/pull/15572#issuecomment-1874099428 @godai78

Comment (godai78) [reply docs/updating/update-to-41.md]: Added to the Update guide.

**Review by arkflpc — APPROVED** [review]

**Pull request merged**